src/server/scripts/Northrend/AzjolNerub/Ahnkahet/boss_amanitar.cpp

```diff
@@ -19,218 +19,298 @@
-            }
+    void JustDied(Unit* /*killer*/) override
+    {
+        _JustDied();
+        DoCastAOE(SPELL_REMOVE_MUSHROOM_POWER);
+        instance->DoRemoveAurasDueToSpellOnPlayers(SPELL_MINI);
+    }
 
-            void SpawnAdds()
-            {
-                uint8 u = 0;
+    void JustSummoned(Creature* summon) override
+    {
+        summons.Summon(summon);
+    }
 
-                for (uint8 i = 0; i < 30; ++i)
-                {
-                    Position pos = me->GetRandomNearPosition(30.0f);
-                    me->UpdateGroundPositionZ(pos.GetPositionX(), pos.GetPositionY(), pos.m_positionZ);
+    void SummonedCreatureDies(Creature* summon, Unit* killer) override
+    {
+        uint32 pos = summon->AI()->GetData(DATA_POSITION);
+        _mushroomsQueue.push(pos);
 
-                    if (Creature* trigger = me->SummonCreature(NPC_TRIGGER, pos))
-                    {
-                        Creature* temp1 = trigger->FindNearestCreature(NPC_HEALTHY_MUSHROOM, 4.0f, true);
-                        Creature* temp2 = trigger->FindNearestCreature(NPC_POISONOUS_MUSHROOM, 4.0f, true);
-                        if (!temp1 && !temp2)
-                        {
-                            u = 1 - u;
-                            me->SummonCreature(u > 0 ? NPC_POISONOUS_MUSHROOM : NPC_HEALTHY_MUSHROOM, pos, TEMPSUMMON_TIMED_OR_CORPSE_DESPAWN, 60 * IN_MILLISECONDS);
-                        }
-                        trigger->DespawnOrUnsummon();
-                    }
-                }
-            }
+        BossAI::SummonedCreatureDies(summon, killer);
+    }
 
-            void UpdateAI(uint32 diff) override
-            {
-                if (!UpdateVictim())
-                    return;
+    void SpawnMushroom(uint32 pos)
+    {
+        if (pos >= MAX_MUSHROOMS)
+            return;
+
+        if (TempSummon* summon = me->SummonCreature(roll_chance_i(40) ? NPC_HEALTHY_MUSHROOM : NPC_POISONOUS_MUSHROOM, MushroomPositions[pos], TEMPSUMMON_CORPSE_TIMED_DESPAWN, 4000))
+            summon->AI()->SetData(DATA_POSITION, pos);
+    }
+
+    void UpdateAI(uint32 diff) override
+    {
+        if (!UpdateVictim())
+            return;
 
-                events.Update(diff);
+        events.Update(diff);
 
-                if (me->HasUnitState(UNIT_STATE_CASTING))
-                    return;
+        if (me->HasUnitState(UNIT_STATE_CASTING))
+            return;
 
-                while (uint32 eventId = events.ExecuteEvent())
-                {
-                    switch (eventId)
+        while (uint32 eventId = events.ExecuteEvent())
+        {
+            switch (eventId)
+            {
+                case EVENT_SPAWN:
+                    for (uint32 i = 0; i < MAX_MUSHROOMS; i++)
+                        SpawnMushroom(i);
+                    break;
+                case EVENT_MINI:
+                    if (Unit* target = SelectTarget(SELECT_TARGET_RANDOM, 0, 0.0f, true, true, -SPELL_MINI))
```

Comment:
This line is giving an error, if you're not gonna use the variable "target" just delete the "Unit* target ="